rust/arrow/src/array/builder.rs

```diff
@@ -86,20 +80,13 @@ pub(crate) fn builder_to_mutable_buffer<T: ArrowPrimitiveType>(
 /// # }
 /// ```
 #[derive(Debug)]
-pub struct BufferBuilder<T: ArrowPrimitiveType> {
+pub struct BufferBuilder<T: ArrowNativeType> {
```

Comment:
This change was motivated by `T` not really requiring `ArrowPrimitiveType`, only `ArrowNativeType`.